pkg/plugin/plugin.go

```diff
@@ -0,0 +1,241 @@
+			if !seen[plugin.Name] {
+				plugins = append(plugins, plugin)
+				seen[plugin.Name] = true
+			}
+		}
+	}
+
+	slices.SortFunc(plugins,
+		func(i, j Plugin) int {
+			return cmp.Compare(i.Name, j.Name)
+		})
+
+	return plugins, nil
+}
+
+func getPluginDirectories() []string {
+	var dirs []string
+
+	selfDirs := usrlocalsharelima.SelfDirs()
+	dirs = append(dirs, selfDirs...)
+
+	pathEnv := os.Getenv("PATH")
+	if pathEnv != "" {
+		pathDirs := filepath.SplitList(pathEnv)
+		dirs = append(dirs, pathDirs...)
+	}
+
+	if libexecDir, err := usrlocalsharelima.LibexecLima(); err == nil {
+		if _, err := os.Stat(libexecDir); err == nil {
+			dirs = append(dirs, libexecDir)
+		}
+	}
+
+	return dirs
+}
+
+// isWindowsExecutableExt checks if the given extension is a valid Windows executable extension
+// according to PATHEXT environment variable.
+func isWindowsExecutableExt(ext string) bool {
+	if runtime.GOOS != "windows" {
+		return false
+	}
+
+	pathExt := os.Getenv("PATHEXT")
+	if pathExt == "" {
+		pathExt = ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL"
+	}
+
+	extensions := strings.Split(strings.ToUpper(pathExt), ";")
+	extUpper := strings.ToUpper(ext)
+
+	for _, validExt := range extensions {
+		if validExt == extUpper {
+			return true
+		}
+	}
+	return false
+}
+
+func scanDirectory(dir string) []Plugin {
+	var plugins []Plugin
+
+	entries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Warnf("Plugin discovery: failed to scan directory %s: %v", dir, err)
+
+		return plugins
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		name := entry.Name()
+		if !strings.HasPrefix(name, "limactl-") {
+			continue
+		}
+
+		pluginName := strings.TrimPrefix(name, "limactl-")
+
+		if strings.Contains(pluginName, ".") {
+			ext := filepath.Ext(name)
+			if !isWindowsExecutableExt(ext) {
+				continue
+			}
+
+			pluginName = strings.TrimSuffix(pluginName, ext)
+		}
+
+		fullPath := filepath.Join(dir, name)
+
+		if !isExecutable(fullPath) {
+			continue
+		}
+
+		plugin := Plugin{
+			Name: pluginName,
+			Path: fullPath,
+		}
+
+		if desc := extractDescFromScript(fullPath); desc != "" {
+			plugin.Description = desc
+		}
+
+		plugins = append(plugins, plugin)
+	}
+
+	return plugins
+}
+
+func isExecutable(path string) bool {
+	info, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+
+	mode := info.Mode()
+	if mode&0o111 != 0 {
+		return true
+	}
+
+	if runtime.GOOS == "windows" {
+		ext := filepath.Ext(path)
+		return isWindowsExecutableExt(ext)
+	}
+
+	return false
+}
+
+func RunExternalPlugin(ctx context.Context, name string, args []string) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	if err := UpdatePathForPlugins(); err != nil {
+		logrus.Warnf("failed to update PATH environment: %v", err)
+		// PATH update failure shouldn't prevent plugin execution
+	}
+
+	externalCmd := "limactl-" + name
+	execPath, err := exec.LookPath(externalCmd)
+	if err != nil {
+		return
+	}
```

Comment:
I think I mentioned this before that you should use the path to the plugin from the discovery code and not call `exec.LookPath`, which might find a different version of the plugin in case there are bugs in the discovery algorithm.

Use the same list in both places to make sure the behaviour is always in sync!